src/schema/posts.ts

```diff
@@ -888,7 +890,16 @@ export const resolvers: IResolvers<any, Context> = {
       ctx,
     ): Promise<GQLEmptyResponse> => {
       await ctx.con.transaction(async (manager) => {
-        await ensureSourcePermissions(ctx, sourceId, SourcePermissions.Post);
+        const source = await ensureSourcePermissions(
+          ctx,
+          sourceId,
+          SourcePermissions.Post,
+        );
+
+        if (source.type === SourceType.Squad) {
+          await canPostToSquadCheck(ctx, source);
+        }
+
```

Comment:
I though about including this logic inside `ensureSourcePermissions` but I think that function already does enough things and we do want to explicitly check this only for squads (for now). So I think this is more verbose.

Reply:
Hmm would argue that would be the better place as it already checks if you are allowed to post.
And it makes it easier down the line if we add more rules to who can post.

Reply:
@rebelchris Issue is that it is not supposed to trigger on every route, only when sharing/creating post/link. To add this I would need to detect current path/mutation inside `ensureSourcePermissions` which I think is another piece of logic that should not be there. Does that make sense?

Reply:
Solved with 4822c4664aa28c883b844ca397980dd1caf41493